learnosity_sdk/request/init.py

```diff
@@ -33,8 +34,8 @@ class Init(object):
     __telemetry_enabled = True
 
     def __init__(
-            self, service, security, secret,
-            request=None, action=None):
+            self, service: str, security: Dict[str, Any], secret: str,
+            request: Dict[str, Any], action:str) -> None:
```

Comment:
<!-- Issues represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion. -->
**🔨 issue:** removing `None` as the default will be a behavioural change in the library behaviour, albeit a very minor one.

<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously. -->
**🎯 suggestion:** I'd suggest we keep the defaults for now for `action` and `request`, while we stick to just keeping the behaviour the same in the meantime?

```python
request: Optional[Dict[str, Any]] = None
```

Reply:
Very true, I've reverted this change and added the Optional flag